proofs/alf/cvc5/Cvc5.smt3

```diff
@@ -128,33 +128,35 @@
   :conclusion (= t (run_evaluate t))
 )
 
-(program is_poly_norm ((T Type) (U Type) (a T) (b T) (a1 U) (a2 U) (b1 U) (b2 U))
-  (T T) Bool
-  (
-    ((is_poly_norm (> a1 a2) (> b1 b2))   (is_poly_norm_arith_mod_c (- a1 a2) (- b1 b2) true))
-    ((is_poly_norm (>= a1 a2) (>= b1 b2)) (is_poly_norm_arith_mod_c (- a1 a2) (- b1 b2) true))
-    ((is_poly_norm (<= a1 a2) (<= b1 b2)) (is_poly_norm_arith_mod_c (- a1 a2) (- b1 b2) true))
-    ((is_poly_norm (< a1 a2) (< b1 b2))   (is_poly_norm_arith_mod_c (- a1 a2) (- b1 b2) true))
-    ((is_poly_norm (= a1 a2) (= b1 b2))   (alf.ite (is_arith_type (alf.typeof a1)) 
-                                          (alf.ite (is_arith_type (alf.typeof b1))
-                                            (is_poly_norm_arith_mod_c (- a1 a2) (- b1 b2) false)
-                                            false)
-                                          false)) ; TODO: bv
-    ((is_poly_norm a b)                   (alf.ite (is_arith_type (alf.typeof a)) 
-                                          (alf.ite (is_arith_type (alf.typeof b))
-                                            (is_poly_norm_arith a b)
-                                            false)
-                                          false)) ; TODO: bv
-  )
-)
-
 ; ProofRule::ARITH_POLY_NORM
 (declare-rule arith_poly_norm ((U Type) (a U) (b U))
   :args ((= a b))
   :requires (((is_poly_norm a b) true))
   :conclusion (= a b)
 )
 
+(program is_poly_norm_rel ((U Type) (r (-> U U Bool)) (cx U) (cy U) (b Bool))
```

Comment:
Use $ as a prefix for any program/define.

I have been incrementally updating this, I don't think I have refactored this part of the signature yet so we should not hit merge conflicts in my other PRs :crossed_fingers:

Reply:
Also, can you add documentation in the style I have been adding (see e.g. https://github.com/cvc5/cvc5/blob/main/proofs/alf/cvc5/programs/Utils.smt3#L100).